torch/distributed/distributed_c10d.py

```diff
@@ -750,9 +745,10 @@ def _check_valid_timeout(timeout: Any) -> None:
 STORE_BASED_BARRIER_PREFIX = "store_based_barrier_key"
 
 
-def _get_pg_default_device(group: Optional[ProcessGroup] = None) -> torch.device:
+def _get_object_coll_device(group: Optional[ProcessGroup] = None) -> str:
     """
-    Return the device to use with ``group`` for control flow usage (object collectives, barrier).
+    Return the device type to use with ``group`` for control flow usage (object
```

Comment:
i'm still not sure i understand what the point of this API is, or how its supposed to be used.  ok let me read the callsites first and come back..

Reply:
ok- iiuc, the issue is that object collective APIs are unique in accepting non-tensor objects which do not have a 'device' attribute to key off of.  Absent that device argument, we look to the PG to determine which device to use.  So having a helper to determine a backend kinda makes sense.  

(or, we could have just added a 'device' argument to obj coll APIs, and made it an error if the user did not specify device and the PG had multiple backends!)

I don't know what "for control flow usage" means either??

Reply:
Yep, that's right. 
Removed the "control flow usage" wording from the description.